_posts/2015-07-17-cassandra-write-performance-on-gce-and-aws.md

```diff
@@ -5,4 +5,87 @@ layout: post
+```
+
+into this:
+```
+juju action do cassandra-stress/0 stress
+```
+
+We ran cassandra-stress with the following configurations:
+
+##A tale of two JDKS: Open JDK vs. Oracle
+
+By default, the Cassandra charm installs OpenJDK 7. We also tested against Oracle’s Java SE 8u51. We did not, however, do any tuning of java parameters. Recommendations from seasoned Cassandra experts would be welcome in this area.
+
+[Marco, I think you had better numbers with this change. I didn’t see much difference between the two on AWS but you did on GCE?]
+
+
+#Results
+##AWS
+
+The best result were were able to achieve on AWS was with a three-node Cassandra cluster on m3.2xlarge instances and a c3.4xlarge instance for `cassandra-stress`.
+
+With that, we were able to achieve 90,748 writes/second with a 99th percentile latency of 4.0ms.
+
+Follow-up tests using r3.4xlarge and i2.4xlarge instances were hit slightly higher numbers but at greater cost. We’ll be looking at how to best benchmark between instance types in a future blog post.
+
+##GCE
+
+The best result we were able to achieve on GCE was 111,394 ops/s and 3ms latency (99th percentile) with a three-node Cassandra cluster using n1-standard-8 instances and a n1-highcpu-16 for `cassandra-stress`.
+
+The next two best results came in at 109,583 ops/s and 101,886 ops/s.
+
+## Summary
+
+With Google's average hourly cost of $0.304 and Amazon's cost of $0.532, Google clearly outperforms it's AWS counterpart at almost half the cost.
+
+All benchmark results for Cassandra are available for comparison [here](http://cloud-benchmarks.org/services/cassandra).
+
+#Lessons Learned
+
+## Building benchmark charms
+We started by adding benchmarking to the Cassandra charm. This sounded ideal; the charm comes packaged with a benchmark. We quickly realized, however, that running `cassandra-stress` on the nodes created a significant resource contention.
+
+We created a new [cassandra-stress](https://jujucharms.com/u/marcoceppi/cassandra-stress/trusty/1) charm to run `cassandra-stress` from its own instance, allowing us to wail on the Cassandra cluster without skewing the results.
+
+## Scaling the benchmark
+As we scaled from three to six nodes in the Cassandra cluster, we unexpectedly saw a slight decrease in performance. We are likely hitting a ceiling with regard to how many ops/sec we can achieve with a single `cassandra-stress` instance.
+
+The next phase of benchmarking Cassandra will look at creating a cluster of `cassandra-stress` machines to simultaneously stress the cluster, as well as determining the optimal number of `cassanda-stress` instances to Cassandra nodes.
+
+## Potential bottlenecks
+There was a negligible differences in performance between r3.2xlarge and r3.4xlarge instances. My suspicion is that we were hitting a bottleneck somewhere between in our environment (network latency, disk i/o, kernel, etc), and we’ll need to isolate that during the next benchmark cycle.
+
+
+
```

Comment:
I'd make the last section "What's next" or soemthing similar where we put our call to actions